src/foundry/client/data/collections/compendium-collection.d.mts

```diff
@@ -390,40 +393,20 @@ declare global {
             doc: StoredDocument<DocumentInstanceForCompendiumMetadata<T>>,
           ) => DeepPartial<DocumentInstanceForCompendiumMetadata<T>["_source"]>),
       condition?: ((obj: StoredDocument<DocumentInstanceForCompendiumMetadata<T>>) => boolean) | null,
-      options?: DocumentModificationContext,
+      options?: DocumentOnUpdateOptions<this["metadata"]["type"]>,
     ): ReturnType<this["documentClass"]["updateDocuments"]>;
 
-    protected _onCreateDocuments(
-      documents: StoredDocument<DocumentInstanceForCompendiumMetadata<T>>[],
-      result: (DocumentInstanceForCompendiumMetadata<T>["_source"] & { _id: string })[],
-      options: DocumentModificationOptions,
-      userId: string,
-    ): void;
-
-    protected _onUpdateDocuments(
-      documents: StoredDocument<DocumentInstanceForCompendiumMetadata<T>>[],
-      result: (DeepPartial<DocumentInstanceForCompendiumMetadata<T>["_source"]> & { _id: string })[],
-      options: DocumentModificationOptions,
-      userId: string,
-    ): void;
-
-    protected _onDeleteDocuments(
-      documents: StoredDocument<DocumentInstanceForCompendiumMetadata<T>>[],
-      result: string[],
-      options: DocumentModificationOptions,
-      userId: string,
-    ): void;
-
-    protected _onDeleteFolder(parentFolder: Folder, deleteFolderId: string, deleteContents?: boolean): string[];
-
     /**
      * Follow-up actions taken when Documents within this Compendium pack are modified
      * @internal
      */
+    // TODO: find the right answer for result
     protected _onModifyContents(
```

Comment:
Looks like this is inherited from `DocumentCollection` and has a different type depending on the DatabaseAction, as this is called by `ClientDatabaseBackend`. 

```js
collection._onModifyContents("create", documents, result, operation, user);
collection._onModifyContents("update", documents, changes, operation, user);
collection._onModifyContents("delete", documents, ids, operation, user);
```
Digging in, it looks like `create` and `update` get `Record<string, unknown>[]`; it's raw serialized document data, while `delete` is `string[]` because it's just the array of ids to delete.

Also, I'm realizing we need to make a decision about these `@internal` marks because they're often _not_ protected, this is called by another class, they're just not intended to be used by community package developers.

Reply:
I committed a change to treat these separately and get the right operation type for each.  Is `Record<string, unknown>` good enough? Or should we refine it based on `T`?    

Would it just be `InstanceType<DocumentClassForCompendiumMetadata<T>>[]`?

Reply:
Record<string, unknown> is sufficient for two reasons
1. This is an internal Foundry method that package developers shouldn't really be messing with
2. There's still potential that this hasn't gone through data model validation